src/lib.rs

```diff
@@ -505,6 +534,70 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight(1000_000)] // TODO weights
+		pub fn clear_expired_match(
+			origin: OriginFor<T>,
+			match_id: T::Hash
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let chess_match = match Self::chess_matches(match_id) {
+				Some(m) => m,
+				None => return Err(Error::<T>::NonExistentMatch.into()),
+			};
+
+			ensure!(
+				(chess_match.state == MatchState::OnGoing(NextMove::Whites)) | 
+				(chess_match.state == MatchState::OnGoing(NextMove::Blacks)), 
+				Error::<T>::MatchNotOnGoing
+			);
+			
+			let now = <frame_system::Pallet<T>>::block_number();
+			let diff = now - chess_match.last_move;
+			let expired: bool = match chess_match.style {
+				MatchStyle::Bullet => diff > T::BulletPeriod::get(),
+				MatchStyle::Blitz => diff > T::BlitzPeriod::get(),
+				MatchStyle::Rapid => diff > T::RapidPeriod::get(),
+				MatchStyle::Daily => diff > T::DailyPeriod::get(),
+			};
+
+			ensure!(expired, Error::<T>::MatchNotExpired);
+
+			let winner = match chess_match.state {
+				MatchState::OnGoing(NextMove::Whites) => chess_match.opponent.clone(),
+				_ => chess_match.challenger.clone(),
+			};
+
+			Self::deposit_event(Event::MatchWon(match_id, winner.clone(), chess_match.board.clone()));
+
+			let time_since_expiry = match chess_match.style {
+				MatchStyle::Bullet => now - (chess_match.last_move + T::BulletPeriod::get()),
+				MatchStyle::Blitz => now - (chess_match.last_move + T::BlitzPeriod::get()),
+				MatchStyle::Rapid => now - (chess_match.last_move + T::RapidPeriod::get()),
+				MatchStyle::Daily => now - (chess_match.last_move + T::DailyPeriod::get()),
+			};
+
+			if (who == chess_match.challenger) | (who == chess_match.opponent) | (time_since_expiry <= T::IncentivePeriod::get()) {
```

Comment:
```suggestion
			let diff = now - chess_match.last_move;
			let abandoned: bool = match chess_match.style {
				MatchStyle::Bullet => diff > T::BulletPeriod::get() * 10u32.into(),
				MatchStyle::Blitz => diff > T::BlitzPeriod::get() * 10u32.into(),
				MatchStyle::Rapid => diff > T::RapidPeriod::get() * 10u32.into(),
				MatchStyle::Daily => diff > T::DailyPeriod::get() * 10u32.into(),
			};

			if (who == chess_match.challenger) | (who == chess_match.opponent) | abandoned {
```

this way we make the incentive period proportional to the dynamics of the match: faster match styles lead to smaller incentive periods... and it's one less `Config` type

Reply:
Yes, I was thinking in the same direction, but wasn't sure which way is better, proportional, or a fixed day for all matches